tests/lifecycle/test_lifecycle_adapters_end_to_end_task_based.py

```diff
@@ -242,6 +246,72 @@ def test_individual_post_task_expand_hook():
     assert len(relevant_calls[0].bound_kwargs["run_id"]) > 10  # Should be UUID(ish)...
 
 
+def test_individual_pre_task_submission_hook():
+    hook_name = "pre_task_submission"
+    hook = TrackingPreTaskSubmissionHook(name=hook_name)
+    dr = _sample_driver(hook)
+    dr.execute(["output"], inputs={"n_iters_input": 5})
+    relevant_calls = [item for item in hook.calls if item.name == hook_name]
+    assert len(relevant_calls) == 10
+    spawning_task_ids = Counter([item.bound_kwargs["spawning_task_id"] for item in relevant_calls])
+    assert spawning_task_ids == {"expand-parallel_over": 5, None: 5}
+    purposes = Counter([item.bound_kwargs["purpose"] for item in relevant_calls])
+    assert purposes == {
```

Comment:
Comments on this might be nice -- E.G. why these are the call counts

Reply:
(although I realize that you probably copied this approach from others which don't document...)

Reply:
Good idea! It took me a few minutes to re-figure out what the counts were looking at them now - a sure sign they are need. Adding now ...